app/1.0/toolbox/news-case-study.md

```diff
@@ -0,0 +1,264 @@
+  </div>
+</div>
+
+## SEO and Social Sharing
+
+A small piece of server code in the News app checks the requesting user agent whenever the user navigates to a URL in your app. If the user agent is a browser, the application returns index.html and the app's routing logic is executed to display the requested content.
+
+However, some user agents (such as certain indexers and social sharing sites) don't run JavaScript. If the user agent is a well-known member of this category, the News app returns a static HTML page so that the page can be parsed and indexed or displayed correctly in a social site.
+
+`main.py`
+```python
+class ArticlePage(webapp2.RequestHandler):
+  def get(self, **kwargs):
+    bots = '|'.join([
+        'Googlebot',
+        'bingbot',
+        'msnbot',
+        'facebookexternalhit',
+        'Facebot',
+        'Twitterbot',
+        'Google-Structured-Data-Testing-Tool'
+      ])
+    if re.search(bots, self.request.headers.get('User-Agent'), re.I):
+      self.response.out.write(open('data/articles/' + kwargs['article_id'] + '.html').read())
+    else:
+      self.response.out.write(open('index.html').read())
+```
+
+## Resource URLs
+
+Resource URLs, like `/bower_components/`, `/data/` and `/src/` are served as static files. These static URLs are defined in `app.yaml`. These URLs are used internally by the app to load components and data.
+
+## Displaying Ads
+
+The News app attempts to display ads, without blocking a fast first paint.
+
+The <news-iframe> element uses [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), a web platform API. `IntersectionObserver` is used to determine whether `<news-iframe>` element is visible on screen before creating an iframe that will display an ad.
+
+`news-iframe.html`
+```javascript
+attached: function() {
+  if ('IntersectionObserver' in window) {
+    this._io = new IntersectionObserver(this._createIframe.bind(this));
+    this._io.observe(this);
+  } else {
+    // Intersection Observer is not available, so just create the iframe
+    // after a bit of a delay.
+    window.setTimeout(this._createIframe.bind(this), 100);
+  }
+},
+```
+
+If `IntersectionObserver` isn't available, the application briefly delays displaying the ad to let the rest of the page load first.
+
+## AMP version of the News app
+
+We built a [version of the News app](https://github.com/PolymerLabs/news/tree/amp) for integration with [AMP (Accelerated Mobile Pages)](https://www.ampproject.org/) documents. In the AMP version of News, the HTML content files (for example, `/data/articles/it-takes-teacher.html`) are replaced by AMP documents which can be loaded as standalone pages.
+
+The AMP project supports adding an AMP document to a shadow root on your page. Here’s an example: <code>[https://github.com/ampproject/amphtml/blob/master/examples/pwa/pwa.js](https://github.com/ampproject/amphtml/blob/master/examples/pwa/pwa.js)</code>
```

Comment:
If you're not going to put a period at the end of this link (which would probably look funky), I'd put it on a separate line.